src/OpenTelemetry.Api/Context/Propagation/AWSXRayPropagator.cs

```diff
@@ -0,0 +1,302 @@
+            var spanId = context.ActivityContext.SpanId.ToHexString();
+            var sampleDecision = (context.ActivityContext.TraceFlags & ActivityTraceFlags.Recorded) != 0 ? SampledValue : NotSampledValue;
+
+            var newTraceHeader = string.Concat(RootKey, KeyValueDelimiter, traceId, TraceHeaderDelimiter, ParentKey, KeyValueDelimiter, spanId, TraceHeaderDelimiter, SampledKey, KeyValueDelimiter, sampleDecision);
+
+            setter(carrier, AWSXRayTraceHeaderKey, newTraceHeader);
+        }
+
+        internal static bool TryParseXRayTraceHeader(string rawHeader, out ActivityContext activityContext)
+        {
+            activityContext = default;
+            string traceId = null;
+            string parentId = null;
+            char traceOptions = default;
+
+            if (string.IsNullOrEmpty(rawHeader))
+            {
+                return false;
+            }
+
+            int position = 0;
+            while (position < rawHeader.Length)
+            {
+                int delimiterIndex = rawHeader.IndexOf(TraceHeaderDelimiter, position);
+                string part;
+                if (delimiterIndex >= 0)
+                {
+                    part = rawHeader.Substring(position, delimiterIndex - position);
+                    position = delimiterIndex + 1;
+                }
+                else
+                {
+                    part = rawHeader.Substring(position);
+                    position = rawHeader.Length;
+                }
+
+                string trimmedPart = part.Trim();
+                int equalsIndex = trimmedPart.IndexOf(KeyValueDelimiter);
+                if (equalsIndex < 0)
+                {
+                    return false;
+                }
+
+                string value = trimmedPart.Substring(equalsIndex + 1);
+
+                if (trimmedPart.StartsWith(RootKey))
+                {
+                    if (!TryParseOTFormatTraceId(value, out var otFormatTraceId))
+                    {
+                        return false;
+                    }
+
+                    traceId = otFormatTraceId;
+                }
+                else if (trimmedPart.StartsWith(ParentKey))
+                {
+                    if (!IsParentIdValid(value))
+                    {
+                        return false;
+                    }
+
+                    parentId = value;
+                }
+                else if (trimmedPart.StartsWith(SampledKey))
+                {
+                    if (!TryParseSampleDecision(value, out var sampleDecision))
+                    {
+                        return false;
+                    }
+
+                    traceOptions = sampleDecision;
+                }
+            }
+
+            if (traceId == null || parentId == null || traceOptions == default)
+            {
+                return false;
+            }
+
+            var activityTraceId = ActivityTraceId.CreateFromString(traceId.AsSpan());
```

Comment:
Lots of string manipulation above to pass a span here. If you switch the above code to manipulate spans over the string/char[] it will probably be less memory instensive/faster overall.

Reply:
Just switched from manipulating `String` to `ReadOnlySpan` in the latest version